packages/core/src/StateNode.ts

```diff
@@ -413,40 +418,10 @@ export class StateNode<
       return undefined;
     }
     if (!nextStateNodes.length) {
-      return {
-        transitions: [selectedTransition],
-        entrySet: [],
-        exitSet: [],
-        configuration: state.value ? [this] : [],
-        source: state,
-        actions
-      };
+      return [selectedTransition];
     }
```

Comment:
Is this being repeated? I think this extra if check is returning the same as the default return value below.

Reply:
Good catch! Yes - this same value is returned from both of those code branches, so we should be able to just return it without introducing a conditional path.